src/openrct2/rct1/S4Importer.cpp

```diff
@@ -1680,7 +1680,7 @@ namespace RCT1
                     dst2->SetStationIndex(StationIndex::FromUnderlying(src2->GetStationIndex()));
                     dst2->SetHasGreenLight(src2->HasGreenLight());
                     dst2->SetIsIndestructible(src2->IsIndestructible());
-                    if (rideType == RIDE_TYPE_GHOST_TRAIN)
+                    if (rideType == EnumValue(RCT1::RideType::GhostTrain))
```

Comment:
I didn’t explain this very well, sorry.

You also need to take rideType from _s4.rides, rather than the imported rides. While at it, also changed line 1708/1709 to check for RCT1::RideType::HedgeMaze rather than retrieving the RTD.